internal/handshake/tls_extension_handler_server.go

```diff
@@ -61,6 +59,29 @@ func (h *extensionHandlerServer) Send(hType mint.HandshakeType, el *mint.Extensi
 	return el.Add(&tlsExtensionBody{data})
 }
 
+// getSupportedVersions returns the supported versions as uint32
+// It adds one reserved version at a random position.
+func (h *extensionHandlerServer) getSupportedVersions() []uint32 {
+	reservedVersion := protocol.GenerateReservedVersion()
+	b := make([]byte, 1)
+	_, _ = rand.Read(b) // ignore the error here. Failure to read random data doesn't break anything
+	randPos := int(b[0]) % (len(h.supportedVersions) + 1)
+	supportedVersions := make([]uint32, len(h.supportedVersions)+1)
+	var j int
+	for _, v := range h.supportedVersions {
+		if j == randPos {
+			supportedVersions[j] = uint32(reservedVersion)
+			j++
+		}
+		supportedVersions[j] = uint32(v)
+		j++
+	}
+	if randPos == len(h.supportedVersions) {
+		supportedVersions[randPos] = uint32(reservedVersion)
+	}
+	return supportedVersions
+}
```

Comment:
This code is fragile enough that I'd like to have it in a single place. WDYT about a single getGreasedVersions in version.go that returns a greased slice?

Reply:
Will be a few more allocations, but probably worth the clarity we gain by doing so.